flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -261,6 +261,7 @@ public void setSavepointRestoreSettings(SavepointRestoreSettings savepointRestor
     public StreamGraph generate() {
         streamGraph = new StreamGraph(executionConfig, checkpointConfig, savepointRestoreSettings);
         shouldExecuteInBatchMode = shouldExecuteInBatchMode(runtimeExecutionMode);
+        streamGraph.setShouldExecuteInBatchMode(shouldExecuteInBatchMode);
```

Comment:
If we're going with the explicit flag, then lets also look into replacing existing usages of `shouldExecuteInBatchMode` to remove some insanity.

Reply:
Alternatively, we could just add a `JobType` to the `StreamGraph`, and set it in `#configureStreamGraph` based on `shouldExecuteInBatchMode`.
With the current structure of the generator this should be the cleaner approach.